assets/js/translator.js

```diff
@@ -868,9 +874,54 @@ function autoSelectDstLang() {
     }
 }
 
+function dictionaryLookup() {
+    if(modeEnabled('lookup')) {
+        if(currentLookupRequest) {
+            currentLookupRequest.abort();
+        }
+
+        $('#dictionaryLookupResult').empty();
+        var wordToLookup = $('#originalText').val().trim()
+            .split(' ');
+        if(wordToLookup.length <= 3 && wordToLookup[0] !== '') {
+            currentLookupRequest = callApy({
+                data: {
+                    'langpair': curSrcLang + '|' + curDstLang,
+                    'q': $('#originalText').val()
+                },
+                success: handleDictionaryLookupSuccessResponse,
+                error: handleDictionaryLookupErrorResponse,
+                complete: function () {
+                    ajaxComplete();
+                    currentLookupRequest = undefined;
+                }
+            }, '/dictionaryLookup');
+        }
+    }
+
+    function handleDictionaryLookupSuccessResponse(data) {
+        $('#dictionaryLookup').removeClass('hide');
+        $('#currentWordForLookup').html($('#originalText').val());
+        for(var analysisType in data) {
+            var values = '<ul id="dictionaryTranslationContents">';
+            for(var translationContent = 0; translationContent < data[analysisType].length; translationContent++) {
+                values += '<li>' + data[analysisType][translationContent].replace('#', '') + '</li>';
+            }
+            values += '</ul>';
```

Comment:
Could you make a `<ul>` first and then append list elements to it? String concatenation is not ideal.